python/vsi/image.py

```diff
@@ -0,0 +1,190 @@
+import math
+from typing import TypeVar, Union, Sequence, Tuple
+from collections.abc import Iterable
+
+import numpy as np
+import numpy.typing as npt
+import scipy.signal
+
+
+# Annotation crap
+T = TypeVar("T", bound=npt.NBitBase)
+
+
+def normalized_cross_correlation_2d(template: "np.floating[T]",
```

Comment:
Why is np.floating[T] in a string?

Reply:
Because numpy told me to?

https://numpy.org/devdocs/reference/typing.html#number-precision